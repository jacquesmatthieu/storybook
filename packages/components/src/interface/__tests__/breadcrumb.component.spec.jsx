import React from 'react';
import { shallow } from 'enzyme';

import Breadcrumb from '../breadcrumb.component';

jest.mock('../../state/withCurrentApp.hoc', () => jest.fn(Comp => Comp));

describe('Breadcrumb', () => {
  let props;

  beforeEach(() => {
    props = {
      steps: ['foo', 'bar'],
      currentApp: { name: 'baz' },
    };
  });

  const getWrapper = () => shallow(<Breadcrumb {...props} />);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
