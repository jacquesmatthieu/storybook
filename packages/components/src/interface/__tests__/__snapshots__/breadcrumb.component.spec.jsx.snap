// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render properly 1`] = `
<breadcrumbcomponent__Bread>
  <breadcrumbcomponent__Crumb
    key="baz-0"
    variant="breadcrumb"
  >
    baz
  </breadcrumbcomponent__Crumb>
  <breadcrumbcomponent__Crumb
    key="foo-1"
    variant="breadcrumb"
  >
    foo
  </breadcrumbcomponent__Crumb>
  <breadcrumbcomponent__Crumb
    key="bar-2"
    variant="breadcrumb"
  >
    bar
  </breadcrumbcomponent__Crumb>
</breadcrumbcomponent__Bread>
`;
